core/src/test/scala/integration/kafka/server/FetcherEventManagerTest.scala

```diff
@@ -98,5 +99,108 @@ class FetcherEventManagerTest {
     fetcherEventManager.close()
   }
 
+  @Test
+  def testEventExecutionThrowsFatalError(): Unit = {
+    val time = Time.SYSTEM
+    val fetcherEventBus = new FetcherEventBus(time)
+    @volatile var truncateAndFetchProcessed = 0
+    val processor : FetcherEventProcessor = new FetcherEventProcessor {
+      override def process(event: FetcherEvent): Unit = {
+        event match {
+          case TruncateAndFetch =>
+            truncateAndFetchProcessed += 1
+            throw new StackOverflowError()
+        }
+
+      }
+
+      override def fetcherStats: FetcherStats = ???
```

Comment:
I'm not familiar with the ??? syntax outside of C escape sequences: what does that do?

Reply:
I did not know before neither. From the source code:

  /** `???` can be used for marking methods that remain to be implemented.
   *  @throws NotImplementedError
   *  @group utilities
   */
  def ??? : Nothing = throw new NotImplementedError

Reply:
Huh, TIL! Thanks, Huilin.